src/CAN/can_mapping.c

```diff
@@ -32,6 +32,8 @@ float canmapping_extract_value(uint64_t raw_data, const CANMapping *mapping)
         }
         uint32_t bitmask = (1UL << length) - 1;
         uint32_t raw_value = (raw_data >> offset) & bitmask;
+
+        /* normalize endian */
```

Comment:
Honestly I would do it now rather than later.  You will totally forget to do it later and the bug will get lost in the noise.  Should be pretty simple to do.  Do something like:
```C
volatile int x = 0xA5;
const char* c = (void*) &x;
switch(*c) {
case 0xA5: return LITTLE_ENDIAN;
case 0x00 : return BIG_ENDIAN;
default; ASSERT(UNREACHABLE);
};
```